crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -92,38 +92,31 @@ impl std::fmt::Display for DisplayRepresentation<'_> {
                 if elements.is_empty() {
                     f.write_str("()")?;
                 } else {
-                    let mut first = true;
-                    for element in &**elements {
-                        if !first {
-                            f.write_str(", ")?;
-                        }
-                        first = false;
-                        element.display(self.db).fmt(f)?;
-                    }
+                    elements.display(self.db).fmt(f)?;
                 }
                 f.write_str("]")
             }
         }
     }
 }
 
-impl<'db> UnionType<'db> {
-    fn display(&'db self, db: &'db dyn Db) -> DisplayUnionType<'db> {
+impl<'a, 'db> UnionType<'db> {
+    fn display(&'a self, db: &'db dyn Db) -> DisplayUnionType<'a, 'db> {
         DisplayUnionType { db, ty: self }
     }
 }
 
-struct DisplayUnionType<'db> {
-    ty: &'db UnionType<'db>,
+struct DisplayUnionType<'a, 'db> {
+    ty: &'a UnionType<'db>,
     db: &'db dyn Db,
 }
 
-impl Display for DisplayUnionType<'_> {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+impl Display for DisplayUnionType<'_, '_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         let elements = self.ty.elements(self.db);
 
         // Group literal types by kind.
-        let mut grouped_literals = FxOrderMap::default();
+        let mut grouped_literals = FxHashMap::default();
```

Comment:
I intentionally used an order map here to preserve the ordering. What's the reason for changing to an fx hash map (and rewriting the formatting here?

Reply:
The ordered properties of the OrderMap were not used at all in the algorithm here. We don't ever iterate over the map; we pop specific keys out of it based on when we run into the first literal of that kind in the overall union.